### PR TITLE
Fix flaky log test

### LINKED ISSUE #8387: Flaky test in test-log.js
`Logging asserts should fail` often flakes, e.g. https://travis-ci.org/ampproject/amphtml/builds/214735215

### PR DESCRIPTION
**Pull request by kmh287**
The deliberately failed assertions would try to report 1% of the time, causing this test to be flaky. For the unit tests in this file, I've replaced the `reportError` function on the window (added in `error.js`) to just return the passed-in error.

Fixes #8387 

/to @choumx

### CODE REVIEW

**Comment by jridgewell** [comment]
Have another fix for this incoming.

**Comment by jridgewell** [comment]
If we go this way, we can remove the other `match`s and `contain`s and instead assert equality.

**Comment by kmh287** [comment]
Actually caught another instance of the same problem with some of the exported functions outside the class. I've replaced the 'contain' calls but would prefer to leave the regexes as is.

File: test/functional/test-log.js [hunk 1]
```diff
@@ -53,12 +53,15 @@ describe('Logging', () => {
         log: logSpy,
       },
       setTimeout: timeoutSpy,
+      reportError: error => error,
     };
+    sandbox.stub(self, 'reportError', error => error);
```
Comment (choumx) [inline test/functional/test-log.js]: Is this necessary? It appears the flaky test uses the fake window object, `win`.

Comment (kmh287) [inline test/functional/test-log.js]: Unfortunately yes. The `rethrowAsync` test uses the real window.

**Pull request merged**